<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1162</width>
    <height>874</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Balrog Control Software</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QGroupBox" name="group_control_panel">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string/>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <widget class="QGroupBox" name="group_connection_handler">
         <property name="title">
          <string>Connect to Test Bench</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QGroupBox" name="group_control_panel_connection_handler_control">
            <property name="title">
             <string/>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <item>
              <widget class="QGroupBox" name="group_control_panel_connection_handler_control_host_port">
               <property name="title">
                <string/>
               </property>
               <layout class="QHBoxLayout" name="horizontalLayout_3">
                <item>
                 <widget class="QLabel" name="label_hostname">
                  <property name="text">
                   <string>Host</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="edit_host">
                  <property name="text">
                   <string>10.42.0.81</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="label_port">
                  <property name="text">
                   <string>Port</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="edit_port">
                  <property name="text">
                   <string>4223</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="groupBox_12">
               <property name="title">
                <string/>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_6">
                <item>
                 <widget class="QPushButton" name="button_connect">
                  <property name="text">
                   <string>Connect</string>
                  </property>
                  <property name="autoDefault">
                   <bool>false</bool>
                  </property>
                  <property name="default">
                   <bool>false</bool>
                  </property>
                  <property name="flat">
                   <bool>false</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="group_connection_status">
            <property name="title">
             <string/>
            </property>
            <layout class="QFormLayout" name="formLayout">
             <item row="0" column="0">
              <widget class="QLabel" name="label_status_connection">
               <property name="text">
                <string>Status</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="label_status_connection_state">
               <property name="text">
                <string>unknown</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_status_hostname">
               <property name="text">
                <string>Hostname</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLabel" name="label_status_hostname_state">
               <property name="text">
                <string>unknown</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="label_status_master_id">
               <property name="text">
                <string>Master ID</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QLabel" name="label_status_master_id_state">
               <property name="text">
                <string>unknown</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_status_port">
               <property name="text">
                <string>Port</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLabel" name="label_status_port_state">
               <property name="text">
                <string>unknown</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="group_preparation">
         <property name="title">
          <string>Preparation</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QGroupBox" name="group_preparation_buttons">
            <property name="title">
             <string/>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <widget class="QPushButton" name="button_selfcheck">
               <property name="text">
                <string>Perform Self Check</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="button_test_horn">
               <property name="text">
                <string>Test Horn</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="button_test_light">
               <property name="text">
                <string>Test Light</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="button_toggle_sensors">
               <property name="text">
                <string>Toggle Sensor</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="button_test_counter">
               <property name="text">
                <string>Test Counter</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="button_test_servo_nitrous_main">
               <property name="text">
                <string>Test Servo Nitrous main</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="button_test_servo_nitrous_vent">
               <property name="text">
                <string>Test Servo Nitrous vent</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="button_test_servo_nitrous_fill">
               <property name="text">
                <string>Test Servo Nitrous fill</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_11">
             <item>
              <widget class="QPushButton" name="button_calibrate_thrust_load">
               <property name="text">
                <string>Calibate thrust load</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="button_calibrate_nitrous_load">
               <property name="text">
                <string>calibrate nitrous load</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="edit_calibrate_load">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="toolTip">
                <string>input kown weight. If empty we use the value 0</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="group_valve_status">
            <property name="title">
             <string>Status</string>
            </property>
            <layout class="QFormLayout" name="formLayout_2">
             <item row="0" column="0">
              <widget class="QLabel" name="label_valve_status_n20_main">
               <property name="text">
                <string>N20 Main Valve</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="label_valve_status_n20_main_state">
               <property name="text">
                <string>unknown</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_valve_status_n20_vent">
               <property name="text">
                <string>N20 Vent Valve</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLabel" name="label_valve_status_n20_vent_state">
               <property name="text">
                <string>unknown</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_valve_status_n20_fill">
               <property name="text">
                <string>N20 Fill Valve</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLabel" name="label_valve_status_n20_fill_state">
               <property name="text">
                <string>unknown</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="label_valve_status_n20_release">
               <property name="text">
                <string>N20 Release Valve</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QLabel" name="label_valve_status_n20_release_state">
               <property name="text">
                <string>unknown</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="group_sequence_loader">
         <property name="title">
          <string>Sequence Loader</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="label_sequence_file_name">
            <property name="text">
             <string>No file selected.</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="group_sequence_buttons">
            <property name="title">
             <string/>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QGroupBox" name="group_sequence_open_reload">
               <property name="title">
                <string/>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_3">
                <item>
                 <widget class="QPushButton" name="button_open_sequence">
                  <property name="text">
                   <string>Open file</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="button_reload_sequence">
                  <property name="text">
                   <string>Reload file</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="group_start_sequence_button">
               <property name="title">
                <string/>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_4">
                <item>
                 <widget class="QPushButton" name="button_start_sequence">
                  <property name="enabled">
                   <bool>true</bool>
                  </property>
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="minimumSize">
                   <size>
                    <width>0</width>
                    <height>5</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>Start Sequence</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="button_abort_sequence">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Abort Sequence</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="button_force_dump">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Force Dump</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1162</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="dockWidget_3">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>500</width>
     <height>148</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::NoDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string comment="Sensor Plots"/>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_5">
    <layout class="QVBoxLayout" name="verticalLayout_8">
     <item>
      <widget class="QGroupBox" name="groupBox_14">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="title">
        <string/>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="PlotWidget" name="plot_pressure_1" native="true"/>
        </item>
        <item row="0" column="1">
         <widget class="PlotWidget" name="plot_pressure_2" native="true"/>
        </item>
        <item row="3" column="0">
         <widget class="PlotWidget" name="plot_load_cell_thrust" native="true"/>
        </item>
        <item row="2" column="1">
         <widget class="PlotWidget" name="plot_pressure_4" native="true"/>
        </item>
        <item row="2" column="0">
         <widget class="PlotWidget" name="plot_pressure_3" native="true"/>
        </item>
        <item row="4" column="0">
         <widget class="PlotWidget" name="plot_thermocouple_nitrous" native="true"/>
        </item>
        <item row="3" column="1">
         <widget class="PlotWidget" name="plot_load_cell_nitrous" native="true"/>
        </item>
        <item row="4" column="1">
         <widget class="PlotWidget" name="plot_thermocouple_engine" native="true"/>
        </item>
        <item row="5" column="0">
         <widget class="PlotWidget" name="plot_differential_pressure" native="true"/>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionOpen_plots_in_new_window">
   <property name="text">
    <string>reset dock widgets</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QWidget</extends>
   <header>pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
